filebeat/input/filestream/prospector.go

```diff
@@ -325,17 +325,21 @@ func (p *fileProspector) onFSEvent(
 	ignoreSince time.Time,
 ) {
 	switch event.Op {
-	case loginp.OpCreate, loginp.OpWrite:
-		if event.Op == loginp.OpCreate {
+	case loginp.OpCreate, loginp.OpWrite, loginp.OpNotChanged:
+		switch event.Op {
+		case loginp.OpCreate:
 			log.Debugf("A new file %s has been found", event.NewPath)
 
 			err := updater.UpdateMetadata(src, fileMeta{Source: event.NewPath, IdentifierName: p.identifier.Name()})
 			if err != nil {
 				log.Errorf("Failed to set cursor meta data of entry %s: %v", src.Name(), err)
 			}
 
-		} else if event.Op == loginp.OpWrite {
+		case loginp.OpWrite:
 			log.Debugf("File %s has been updated", event.NewPath)
+
+		case loginp.OpNotChanged:
+			log.Debugf("File %s has not changed, trying to start new harvester because 'delete' is enabled", event.NewPath)
```

Comment:
There is no check that delete is enabled here. This log line depends on the conditions where OpNotChanged is sent never changing, but that logic is in a totally separate file where someone changing it would not know to consider this line here.

Reply:
That is true, thanks for pointing it out.

I'll re-write it to only state what is happening and not why (delete feature enabled). So it will always be consistent.